### PR TITLE
Add BaseItemManager

### PR DESCRIPTION
**Pull request by crobibero**
Also move `IsMetadataFetcherEnabled` and `IsImageFetcherEnabled`

### CODE REVIEW

File: Jellyfin.Server/CoreAppHost.cs [hunk 1]
```diff
@@ -76,6 +77,7 @@ protected override void RegisterServices()
                  options => options.UseSqlite($"Filename={Path.Combine(ApplicationPaths.DataPath, "jellyfin.db")}"));
 
             ServiceCollection.AddEventServices();
+            ServiceCollection.AddScoped<IBaseItemManager, BaseItemManager>();
```
Comment (barronpm) [inline Jellyfin.Server/CoreAppHost.cs:80]: This should be a singleton until everything that depends on it is made scoped

**Review by barronpm — APPROVED** [review]

File: MediaBrowser.Controller/BaseItemManager/BaseItemManager.cs [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        {
+            _serverConfigurationManager = serverConfigurationManager;
+        }
+
+        /// <inheritdoc />
+        public bool IsMetadataFetcherEnabled(BaseItem baseItem, LibraryOptions libraryOptions, string name)
+        {
+            if (baseItem is Channel)
+            {
+                // hack alert
```
Comment (BaronGreenback) [inline MediaBrowser.Controller/BaseItemManager/BaseItemManager.cs:29]:
```suggestion
                // Hack alert.
```

File: MediaBrowser.Controller/BaseItemManager/BaseItemManager.cs [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+                return true;
+            }
+
+            if (baseItem.SourceType == SourceType.Channel)
+            {
+                // hack alert
```
Comment (BaronGreenback) [inline MediaBrowser.Controller/BaseItemManager/BaseItemManager.cs:35]:
```suggestion
                // Hack alert.
```

File: MediaBrowser.Controller/BaseItemManager/BaseItemManager.cs [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+            return itemConfig == null || !itemConfig.DisabledImageFetchers.Contains(name, StringComparer.OrdinalIgnoreCase);
+        }
+
+        /// <inheritdoc />
+        public bool IsImageFetcherEnabled(BaseItem baseItem, LibraryOptions libraryOptions, string name)
+        {
+            if (baseItem is Channel)
+            {
+                // hack alert
```
Comment (BaronGreenback) [inline MediaBrowser.Controller/BaseItemManager/BaseItemManager.cs:60]:
```suggestion
                // Hack alert.
```

File: MediaBrowser.Controller/BaseItemManager/BaseItemManager.cs [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+                return true;
+            }
+
+            if (baseItem.SourceType == SourceType.Channel)
+            {
+                // hack alert
```
Comment (BaronGreenback) [inline MediaBrowser.Controller/BaseItemManager/BaseItemManager.cs:66]:
```suggestion
                // Hack alert.
```

**Review by BaronGreenback — CHANGES_REQUESTED** [review]

**Review by BaronGreenback — APPROVED** [review]

**Pull request merged**